storage.go

```diff
@@ -218,14 +218,17 @@ func (storage *MongoStorage) UpdateJob(job *SubmittedJob) error {
 // GetAccount loads an account by its unique account name, creating it if it doesn't already exist.
 func (storage *MongoStorage) GetAccount(name string) (*Account, error) {
 	out := Account{Name: name}
-	_, err := storage.accounts().FindId(name).Apply(mgo.Change{
-		Update:    bson.M{"$setOnInsert": out},
-		Upsert:    true,
-		ReturnNew: true,
-	}, &out)
-	if err != nil {
+	var acct Account
+	err := storage.accounts().FindId(name).One(&acct)
+	if err != nil && err != mgo.ErrNotFound {
 		return nil, err
 	}
+	if err == mgo.ErrNotFound {
+		err = storage.accounts().Insert(out)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
The idea is that we can scale this service horizontally - run a ton of cloudpipe containers and load-balance among them. I could easily see two requests coming in from the same user being routed to two instances of this service and having them fight.

A transaction would be :+1: too.